### ISSUE TITLE
Allow to stop popping test if there already X failures

### ISSUE THREAD

**Issue by casperisfine** [Shopify/ci-queue#33]
In some extreme cases, a single low level issue can make a huge portion of the test suite fail. 

It can make sense in this context to stop running tests once X errors have been recorded, both to save CPU time and to keep a readable output.

cc @Shopify/pipeline

**Comment by casperisfine** [comment]
NB: the tricky part is for the summary job to realize all workers gave up (when it happens).

**Comment by wvanbergen** [comment]
We can use the new `wait` feature of BK, and only start the summarize job after all workers have finished even if some of them fail.

**Comment by wvanbergen** [comment]
After #68, do we still need to do something for the summarize job?

**Comment by casperisfine** [comment]
@DazWorrall suggested we could detect such issue by keeping track of the state of the queue. If nothing was popped for longer than the queue timeout then we could assume all workers are gone.

I think it would workin most case, but it's a bit finicky, because the queue could have kept the same size but test could have been requeued. It's quite a corner case though.

Alternatively we could have all workers write `Time.now` in a single key whenever they reach to Redis, this way the `report` command can see the last time a worker reported back.

In any case, I think It's fine to ship #68 without that safety in place. The report process waiting on dead workers is already an issue (e.g. all workers crash while loading the app), so I think we should treat it as a separate issue.

**Comment by wvanbergen** [comment]
:+1: in the end the job timeout will fail the summary task as well
